### PR TITLE
SNOW-652674: normalize name and denormalize name should handle empty string

### PR DESCRIPTION
**Pull request by AdGnil**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-652674

trying to fix issue: https://github.com/snowflakedb/snowflake-sqlalchemy/issues/340, and based on discussion in https://github.com/sqlalchemy/sqlalchemy/discussions/8444.

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

column name could be an empty string, and to fix it, we should check if string is empty and return empty string accordingly when normalizing/denormalizing names.

### CODE REVIEW

**Comment by AdGnil** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

File: tests/test_core.py [hunk 1]
```diff
@@ -1718,3 +1718,45 @@ def test_result_type_and_value(engine_testaccount):
...
+        )
+        results = conn.exec_driver_sql(
+            f"""
+SELECT * FROM {table_name} UNPIVOT(SALES FOR "" IN (JAN, FEB))  ORDER BY EMPID;"""
+        ).fetchall()  # normalize_name will be called
+        assert results == [
+            (1, "electronics", "JAN", 100),
+            (1, "electronics", "FEB", 200),
+            (2, "clothes", "JAN", 100),
+            (2, "clothes", "FEB", 300),
```
Comment (sfc-gh-aling) [inline tests/test_core.py:1747]:
```suggestion
            (1, "ELECTRONICS", "JAN", 100),
            (1, "ELECTRONICS", "FEB", 200),
            (2, "CLOTHES", "JAN", 100),
            (2, "CLOTHES", "FEB", 300),
```

**Review by sfc-gh-aling — APPROVED** [review]

**Comment by sfc-gh-aling** [comment]
need to create a separate PR to run test as secrets can not be passed from a different fork: https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

**Pull request closed without merging**